app/controllers/macros_controller.rb

```diff
@@ -1,8 +1,12 @@
 class MacrosController < ApplicationController
-  before_action :authorize_deployer!
-  before_action :authorize_admin!, only: [:new, :create, :edit, :update, :destroy]
+  include CurrentProject
+  include ProjectLevelAuthorization
 
-  before_action :find_project
+  before_action :authorize_deployer!
+  before_action do
+    find_project(params[:project_id])
+  end
```

Comment:
In this PR, `authorize_deployer` is only checking if the User is a system level deployer, so, it doesn't require the project being loaded. That's only required by the `authorize_project_admin`.